### PR TITLE
[0.64] NuGet Fix for CI

### PR DESCRIPTION
**Pull request by AgneLukoseviciute**
Backporting #7544

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/7548)

### CODE REVIEW

File: .ado/templates/e2e-test-job.yml [hunk 1]
```diff
@@ -28,6 +28,29 @@ jobs:
             
       - template: prepare-env.yml
 
+      - task: CmdLine@2
```
Comment (dannyvv) [inline .ado/templates/e2e-test-job.yml:31]: Could we try 0.64 without all these nuget commands? Maybe just the nuget.config files?
That way we know that we can clean them in in the main branch too..

Comment (AgneLukoseviciute) [reply .ado/templates/e2e-test-job.yml:31]: I tried that with the initial commit thinking the same thing, seems like adding the nuget.congifs alone was not enough. I am curious though if `nuget sources list` before the restore is enough on it's own, I'll try removing the cache and solution cleans.

**Comment by dannyvv** [comment]
It seems 64 might need @asklar 's fix to always write the nuget.config file in `react-native-windows-init`

**Review by NickGerleman — APPROVED** [review]

File: .ado/ConditionalNugetAdd.ps1 [hunk 2]
```diff
@@ -0,0 +1,5 @@
+$sources = nuget sources list
+$found? = $sources -Match "https://api.nuget.org/v3/index.json"
```
Comment (JunielKatarn) [inline .ado/ConditionalNugetAdd.ps1:2]: For my education, why is this variable sufficed with a `?`?
Does this make it nullable/optional?
If so, why does it need to be nullable/optional?

Comment (AgneLukoseviciute) [reply .ado/ConditionalNugetAdd.ps1:2]: @JunielKatarn just part of the variable name, old habits die hard I guess.. shouldn't have any effect, but I might rename/get rid of `?` if it's confusing.

**Pull request merged**